faker/providers/internet/__init__.py

```diff
@@ -68,7 +69,7 @@ def _to_ascii(self, string):
         for search, replace in self.replacements:
             string = string.replace(search, replace)
 
-        return string
+        return string.encode('ascii', 'ignore').decode('utf8')
```

Comment:
👍

Reply:
@fcurella:

I removed the uuid4 emails (unfortunately, our comments were gone with the change too).

Reply:
I would add a `drop_ascii = True` property on this class, check for its value in the method, and set the property to `False` for the providers in `/providers/internet/ar_*`

Reply:
@fcurella:

It could be done the same way it's done for de_DE. So, something like this, but then for Arabic:

```python

# coding=utf-8

from __future__ import unicode_literals
from .. import Provider as InternetProvider

class Provider(InternetProvider):

    free_email_domains = (
        'aol.de', 'gmail.com', 'gmx.de', 'googlemail.com', 'hotmail.de',
        'web.de', 'yahoo.de',
    )
    tlds = ('com', 'com', 'com', 'net', 'org', 'de', 'de', 'de', )

    replacements = (
        ('ä', 'ae'), ('Ä', 'Ae'),
        ('ö', 'oe'), ('Ö', 'Oe'),
        ('ü', 'ue'), ('Ü', 'Ue'),
        ('é', 'e'), ('É', 'E'),
        ('à', 'a'), ('À', 'A'),
        ('ß', 'ss'),
    )
```

Reply:
that would be the best way, but I don't know how to transliterate arabic (assuming it's even possible)

Reply:
@fcurella:

I've just done that. See the updated tests and the ``ar_AA module``.

Reply:
@fcurella:

Question. Since you know faker better, we always transliterate **to** ASCII and never **from** ASCII, right? If so, there is a more elegant solution, using a dependency (``unidecode``), which does it pretty well for (almost) all languages.

Reply:
Just a snippet of code:

```python

import unidecode

unidecode.unidecode('اصيل')
unidecode.unidecode('Ժամանակով Կատուն ճոն էր')
unidecode.unidecode('У попа была собака')
```

Reply:
the problem with using `unidecode` is that `unidecode` is not locale aware. For example, it converts `ü` to `u`, when in German it should be replaced with `ue`.

I think we can combine the two: use the replacement list if it's defined, or fallback to unidecode if it's not.